### PR TITLE
Config plugins v2

### PR DESCRIPTION
**Pull request by jaybosamiya**
This PR updates the remaining plugins to use the `Config` interface.

### CODE REVIEW

**Comment by jaybosamiya** [comment]
Minor issues:
+ The api plugin's help will now show `--rem` and `--remove` as separate options, since we do not have alias abilities in our `Config` interface. The reason for not allowing aliases was to have a single point of truth at the command line for the plugins. However, the reason for aliasing here is to ensure that people's scripts don't break.
+ The phoenix plugin's `cfg_format` code feels a little weird to me. Might want to look into improving it later on.

File: plugins/phoenix/phoenix_main.ml [hunk 1]
```diff
@@ -17,56 +16,43 @@ let run options project =
...
+  let doc = sprintf "Put %s on graph labels" y in
+  Config.(flag name ~doc)
 
-let output_folder : _ Term.t =
-  let doc = "Output data into the specified folder" in
-  let default = "phoenix" in
-  Arg.(value & opt string default & info ["output-folder"] ~doc)
+let with_name = label_with "name" "block name"
+let with_asm  = label_with "asm"  "assembler instructions"
+let with_bil  = label_with "bil"  "bil instructions"
```
Comment (ivg) [inline plugins/phoenix/phoenix_main.ml]: there is a semantic difference here, in the base the order in which options were specified was preserved, and this allowed us to specify format `name, asm, bil` or `name, bil, asm` etc.

File: plugins/phoenix/phoenix_main.ml [hunk 2]
```diff
@@ -76,10 +62,23 @@ let man = [
...
+  Config.manpage man;
+  Config.when_ready (fun {Config.get=(!)} ->
+      let cfg_format =
+        let self = [] in
+        let self = if !with_name then `with_name :: self else self in
+        let self = if !with_asm  then `with_asm  :: self else self in
+        let self = if !with_bil  then `with_bil  :: self else self in
+        match self with
+        | [] -> [`with_name]
+        | _ as x -> x in
```
Comment (ivg) [inline plugins/phoenix/phoenix_main.ml]: it is not the choice of one from three, it should work as a checkbox basically. Something like this:
```ocaml
let cfg_format = List.filter [`with_name,!with_name; `with_asm, !with_asm; `with_bil, !with_bil] ~f:snd |> List.map ~f:fst
```

File: plugins/phoenix/phoenix_main.ml (hunk 1 above)
Comment (jaybosamiya) [inline plugins/phoenix/phoenix_main.ml]: Well, with the current `Config` interface then we cannot preserve this, and will have to extend our interface.

**Comment by jaybosamiya** [comment]
As I mentioned in my comment,
> The reason for not allowing aliases was to have a single point of truth at the command line for the plugins

**Comment by ivg** [comment]
> As I mentioned in my comment

yeah, I noticed it, but we still have synonyms. On the other hand, cmdliner accepts any unambiguous prefix of an option, so that means that if you have `--api-remove` then `--api-r` and `--api-re` and `--api-rem` are all implicit synonyms of `--api-remove`. So in this PR you don't need to specify option twice, just removing `rem` variant will not break things.

But, I still think that there is nothing wrong with synonyms, so we can add an optional `string list` flag for it. A user will be discouraged from using it as he needs to write `~synonyms:["a-better-name"]`. So he will use it only if it really have a reason to do this.

Also, back to our offline discussion, I think that the better type for the deprecated option would string. So that a user can write:
```ocaml
Option.(param string "output.txt" ~deprecated:"use our new database interface instead")
```

**Comment by jaybosamiya** [comment]
The following changes have been made now:

+ Merge master (to be able to use features from #505)
+ Use synonyms for `--api-remove` to also accept `--api-rem`
+ Start showing deprecation warnings for `--labels-with-*` and favour `--labels-with=`
+ Cleaned up the code that is used in the deprecation part.

**Pull request merged**